performance-tests/bench_2/worker/WriteAction.cpp

```diff
@@ -173,10 +182,9 @@ void WriteAction::stop()
   if (started_ && !stopped_) {
     stopped_ = true;
     proactor_.cancel_timer(*handler_);
-    const DDS::Duration_t timeout = { 0, 0 };
-    data_dw_->wait_for_acknowledgments(timeout);
+    data_dw_->wait_for_acknowledgments(final_wait_for_ack_);
     data_dw_->unregister_instance(data_, instance_);
-    data_dw_->wait_for_acknowledgments(timeout);
+    data_dw_->wait_for_acknowledgments(final_wait_for_ack_);
```

Comment:
I'm curious why `wait_for_acknowledgments` is called twice here?

Reply:
To allow data to propagate before unregistering the instance (potentially gapping data that needs to be resent), and then to allow the instance unregister to propagate before exiting. In general, the instance stuff in WriteAction probably needs a bit of work. I was just lazy here and didn't want to subdivide the final_wait_for_ack_ time period manually here so it uses the same period twice.